backend/src/main/java/com/google/blackswan/mock/SimpleRelatedDataGenerator.java

```diff
@@ -0,0 +1,119 @@
+
+/** Singleton class to generate related data for a given anomaly. */
+public class SimpleRelatedDataGenerator implements RelatedDataGenerator {
+  private static final String TARGET_DIMENSION = "Ramen";
+  private static final String RELATED_METRIC = "Interest Over Time";
+  private static final String CONFIG_USERNAME = "catyu@";
+  private static final ImmutableMap<String, String> FILE_LOCATION = ImmutableMap.of(
+                                                                      "Udon", "/udon-data.csv",
+                                                                      "Pho", "/pho-data.csv"
+                                                                    );
+
+  private static final SimpleRelatedDataGenerator INSTANCE = new SimpleRelatedDataGenerator();
+
+  private Multimap<String, DataInfo> relatedDataMap;
+  private Map<String, Map<Timestamp, Integer>> csvDataCache;
+
+  private SimpleRelatedDataGenerator() {
+    relatedDataMap = ArrayListMultimap.create();
+    csvDataCache = new HashMap<>();
+    prefillRelatedData(); // TODO: Remove when connected to datastore.
+  }
+
+  /** 
+   * Simulates action of querying configs from datastore and adding to 
+   * relatedDataMap. Currently, manually adds these relationship.
+   */
+  private void prefillRelatedData() {
+    // TODO: Replace with call for querying configs from datastore.
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Udon", CONFIG_USERNAME));
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Pho", CONFIG_USERNAME));
+    // Multimap looks like this right now: {{Ramen,{Udon, Pho}}, ...}.
+  }
+
+  public List<RelatedData> getRelatedData(String metricName, String dimensionName, 
+      Timestamp startTime, Timestamp endTime) {
+    List<RelatedData> relatedDataList = new ArrayList<RelatedData>();
+    // TODO: Currently only tries to find related data for dimension, ex. Ramen -> Pho, 
+    //       does not include logic for metric, should be similar.
+    if (relatedDataMap.containsKey(dimensionName)) {
+      for (DataInfo relatedTopic : relatedDataMap.get(dimensionName)) {
+        Map<Timestamp, Integer> topicDataPoints = 
+            getTopicDataPoints(relatedTopic.getDimensionName());
+
+        Map<Timestamp, MetricValue> dataPointsPlot = 
+            getDataPointsInRange(topicDataPoints, startTime, endTime);
+        if (dataPointsPlot.isEmpty()) {
+          continue;
+        }
+
+        relatedDataList.add(new RelatedData(
+            relatedTopic.getUsername(), relatedTopic.getMetricName(), 
+            relatedTopic.getDimensionName(), dataPointsPlot)
+          );        
+      }
+    }
+    return relatedDataList;
+  }
+
+  private ImmutableMap<Timestamp, Integer> getTopicDataPoints(String topic) {
+    if (!csvDataCache.containsKey(topic)) {
+      csvDataCache.put(topic, CSVParser.parseCSV(
+        SimpleRelatedDataGenerator.class.getResourceAsStream(FILE_LOCATION.get(topic))
```

Comment:
so this is getting all of the data from the CSV file, correct?
also see comment about changing to FILE_LOCATIONS

Reply:
yes, that is correct!